src/htmx.js

```diff
@@ -1923,10 +1919,10 @@ return (function () {
 
         function findElementsToProcess(elt) {
             if (elt.querySelectorAll) {
-                var boostedElts = hasChanceOfBeingBoosted() ? ", a" : "";
-                var results = elt.querySelectorAll(VERB_SELECTOR + boostedElts + ", form, [type='submit'], [hx-sse], [data-hx-sse], [hx-ws]," +
+                var boostedElts = document.querySelectorAll("[hx-boost] a, [data-hx-boost] a, a[hx-boost], a[data-hx-boost]");
+                var results = elt.querySelectorAll(VERB_SELECTOR + ", form, [type='submit'], [hx-sse], [data-hx-sse], [hx-ws]," +
                     " [data-hx-ws], [hx-ext], [data-hx-ext], [hx-trigger], [data-hx-trigger], [hx-on], [data-hx-on]");
-                return results;
+                return toArray(results).concat(toArray(boostedElts));
```

Comment:
Now that I see it, couldn't you just add the hx-boost selectors to `result`'s querySelectorAll, and avoid having to concat the arrays altogether as you would retrieve all elements with a single `querySelectorAll` call?